### PR TITLE
Arifnyet switch/checkbox   default value

### PR DESCRIPTION
**Pull request by arifnyet**
Pr for Set default value for list filter widget #2329

### CODE REVIEW

File: modules/backend/widgets/Filter.php [hunk 1]
```diff
@@ -566,7 +566,11 @@ protected function makeFilterScope($name, $config)
         /*
          * Set scope value
          */
-        $scope->value = $this->getScopeValue($scope);
+        if($scopeType=='checkbox' || $scopeType=='switch'){
```
Comment (LukeTowers) [inline modules/backend/widgets/Filter.php]: Please use strict type comparisons (`===`), not loose type comparisons.

File: modules/backend/widgets/Filter.php [hunk 2]
```diff
@@ -566,7 +566,11 @@ protected function makeFilterScope($name, $config)
...
+            $scope->value = isset($config['value']) ? $config['value'] : $this->getScopeValue($scope);
```
Comment (LukeTowers) [inline modules/backend/widgets/Filter.php]: The config option should be `'default'`, not `'value'`. Also, does this prevent the user from overriding the default? It seems like if a default is defined then the value retrieved from `$this->getScopeValue($scope)` would not be used.

File: modules/backend/widgets/filter/partials/_scope_checkbox.htm [hunk 3]
```diff
@@ -2,6 +2,6 @@
 <div
     class="filter-scope checkbox custom-checkbox"
     data-scope-name="<?= $scope->scopeName ?>">
-    <input type="checkbox" id="<?= $scope->getId() ?>" <?= $scope->value ? 'checked' : '' ?> />
+    <input type="checkbox" id="<?= $scope->getId() ?>" <?= $scope->value==1 ? 'checked' : '' ?> />
```
Comment (LukeTowers) [inline modules/backend/widgets/filter/partials/_scope_checkbox.htm]: Again, strict type comparison please. Also, why is this check necessary?

File: modules/backend/widgets/Filter.php (hunk 1 above)
Comment (arifnyet) [reply modules/backend/widgets/Filter.php]: @LukeTowers  done

File: modules/backend/widgets/Filter.php (hunk 2 above)
Comment (arifnyet) [reply modules/backend/widgets/Filter.php]: @LukeTowers  done

File: modules/backend/widgets/filter/partials/_scope_checkbox.htm (hunk 3 above)
Comment (arifnyet) [reply modules/backend/widgets/filter/partials/_scope_checkbox.htm]: @LukeTowers  it is not necessary.. done changing

**Comment by arifnyet** [comment]
@LukeTowers  updated the pr

**Comment by LukeTowers** [comment]
Did this address the issue with the default option overriding the user selected one all of the time?

**Comment by arifnyet** [comment]
@LukeTowers sorry but i think this method is redundant with session caches.... im working on new method i close this pr for now... i submit a new one when it is ready

**Pull request closed without merging**